lib/Dialect/FIRRTL/Transforms/InferResets.cpp

```diff
@@ -890,10 +888,11 @@ void InferResetsPass::traceResets(FIRRTLType dstType, Value dst, unsigned dstID,
                        srcID, loc);
   }
 
-  auto dstBase = dstType.cast<FIRRTLBaseType>();
-  auto srcBase = srcType.cast<FIRRTLBaseType>();
-  if (dstBase.isGround()) {
-    if (dstBase.isa<ResetType>() || srcBase.isa<ResetType>()) {
+  auto dstBase = dstType.dyn_cast<FIRRTLBaseType>();
+  auto srcBase = srcType.dyn_cast<FIRRTLBaseType>();
+  if (!dstBase || dstBase.isGround()) {
```

Comment:
If this fails, this will get dropped into the `llvm_unreachable`, yes? I think the behavior is to just ignore it which is slightly different. Also, can this be reordered to early exit to remove nesting? 😉

Reply:
Yeah arguably that unreachable is not really useful. This should likely just be a check for reset types and corresponding handling of reset types. And if they're not reset types, just don't do anything.